### DISCUSSION TITLE
how to add filtering within the category and product filters

### DISCUSSION THREAD

**Discussion by yellotra** [OFFLINE-GmbH/oc-mall-plugin#1039] [Q&A]
Hi! How to filter products to show only products for the category and used product filters for pagination:

$posts = Product::published()->paginate(12, post('page')); - works, but not filtered

$posts = Product::published()->('category_id',  $categoryId)->paginate(12, post('page')); - does not work

**Comment by tobias-kuendig** [comment]
You code is missing the `where`:

```php
$posts = Product::published()
    ->where('category_id', $categoryId)
    ->paginate(12, post('page')); 
```

Also please specify what "does not work" means exactly (what error? what wrong behaviour?), if the code above does not fix your problem.

**Reply by yellotra** [reply]
Thank you for the answer. I want to do infinite scroll of "load more" buttom for pagintation.
I used two solutions but they don't work correctly.

1). https://github.com/OFFLINE-GmbH/oc-mall-plugin/pull/781 - because of the bugs (products are duplicated and bugs come out after 3td scroll).
2). I use following code in products.php:

    public function onLoadMore()
    {

      $items = $this->getItems();

      $nextpage = $items->hasMorePages()? post('page') + 1 : '0';

      return [
        'partial' => $this->renderPartial($this."::make_posts", ['items' => $item]),
        'loadmore' => $posts->hasMorePages(),
        'nextpage' => $nextpage
      ];
    }

And ajax infinite scroll for this function
    
but this code does not work with filters
